yt_dlp/extractor/nhk.py

```diff
@@ -23,96 +23,56 @@
-                r'prod:[^;]+\bapiUrl:\s*[\'"]([^\'"]+)[\'"]', movie_player_js, None, 'stream API url'),
-            'token': self._search_regex(
-                r'prod:[^;]+\btoken:\s*[\'"]([^\'"]+)[\'"]', movie_player_js, None, 'stream API token'),
-        }
-        self.cache.store('nhk', 'api_info', api_info)
-        return api_info
-
-    def _extract_stream_info(self, vod_id):
-        for refresh in (False, True):
-            api_info = self._get_api_info(refresh)
-            if not api_info:
-                continue
-
-            api_url = api_info.pop('url')
-            meta = traverse_obj(
-                self._download_json(
-                    api_url, vod_id, 'Downloading stream url info', fatal=False, query={
-                        **api_info,
-                        'type': 'json',
-                        'optional_id': vod_id,
-                        'active_flg': 1,
-                    }), ('meta', 0))
-            stream_url = traverse_obj(
-                meta, ('movie_url', ('mb_auto', 'auto_sp', 'auto_pc'), {url_or_none}), get_all=False)
-
-            if stream_url:
-                formats, subtitles = self._extract_m3u8_formats_and_subtitles(stream_url, vod_id)
-                return {
-                    **traverse_obj(meta, {
-                        'duration': ('duration', {int_or_none}),
-                        'timestamp': ('publication_date', {unified_timestamp}),
-                        'release_timestamp': ('insert_date', {unified_timestamp}),
-                        'modified_timestamp': ('update_date', {unified_timestamp}),
-                    }),
-                    'formats': formats,
-                    'subtitles': subtitles,
-                }
-        raise ExtractorError('Unable to extract stream url')
+        content_format = 'video' if is_video else 'audio'
+
+        if not is_episode:
+            page_type = 'programs'
+
+        if is_clip:
+            content_type = 'clips'
+        else:
+            content_type = 'episodes'
+
+        if not is_episode:
+            extra_page = f'/{content_format}_{content_type}'
+        else:
+            extra_page = ''
+            page_type = content_type
+
+        return self._download_json(self._API_URL_TEMPLATE.format(
+            lang=lang,
+            content_format=content_format,
+            page_type=page_type,
+            m_id=m_id,
+            extra_page=extra_page,
+        ), video_id=join_nonempty(m_id, lang),
+        )
+
+        # FOR EXAMPLE:
+        # https://api.nhkworld.jp/showsapi/v1/en/video_episodes/2024117
+        # https://api.nhkworld.jp/showsapi/v1/en/audio_episodes/r_l_japannews-20250203-1
+        # https://api.nhkworld.jp/showsapi/v1/en/video_clips/9999011
+        # https://api.nhkworld.jp/showsapi/v1/en/video_programs/dwc/video_episodes
+        # https://api.nhkworld.jp/showsapi/v1/en/audio_programs/livinginjapan/audio_episodes
+        # no such thing as an audio_clips afaict
```

Comment:
I don't understand this comment, so it's not helpful here IMO. Add them as tests if you need them as examples
```suggestion
```